src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -164,13 +173,32 @@ class ContentExplorer extends Component {
     }
 
     componentDidUpdate({ initialFoldersPath: prevInitialFoldersPath }) {
+        const { allItemsEnabled, folderToDeepScan, includeSubfolders, itemsAreSelectedAndActionDisabled } = this.state;
         const { initialFoldersPath } = this.props;
 
         if (prevInitialFoldersPath !== initialFoldersPath) {
             this.setState({
                 foldersPath: initialFoldersPath,
             });
         }
+
+        // If we are inside a selected folder while the include subfolders toggle is on
+        if (includeSubfolders && !itemsAreSelectedAndActionDisabled && this.shouldSelectAndActionDisableItems()) {
+            this.setState({ itemsAreSelectedAndActionDisabled: true, allItemsEnabled: false });
+        }
+        // If we turn include subfolders toggle off while inside a folder that was chosen to include subfolders
+        else if (itemsAreSelectedAndActionDisabled && !this.shouldSelectAndActionDisableItems()) {
+            this.setState({ itemsAreSelectedAndActionDisabled: false });
+        }
+        // If we return back to the folder that our folder chosen for including subfolders is in
```

Comment:
```suggestion
        // If we return back to the folder that our folder-chosen-for-including-subfolders is in
```

Reply:
Assuming I got it right. If not, please make it more clear.

Reply:
No yeah that is what I was going for. Thank you for the suggestion